src/Types/RejectReason.cs

```diff
@@ -0,0 +1,421 @@
+using Concordium.Sdk.Exceptions;
+using Google.Protobuf;
+
+namespace Concordium.Sdk.Types;
```

Comment:
Is it not possible to have `RejectReason<T> where T : RejectReason<T>`? This will carry type information of the reject reason, no?

Reply:
Not sure I’m follow - can you give an example?

Reply:
So currently, we need to map each enum variant to a field declared in this class, no?

So if we had something like `InvalidEncryptedAmountTransferProof : RejectReason<InvalidEncryptedAmountTransferProof>`, then the `ReasonCase` field could be factored into the abstract `RejectReason<T>` class and we would not have to use this mapping to determine the non-null fields, since `InvalidEncryptedAmountTransferProof` would have only the fields specific to it?